content/ui-testing-handbook/react/en/composition-testing.md

```diff
@@ -0,0 +1,244 @@
+Default.args = {
+  tasks: [
+    { id: '1', state: 'TASK_INBOX', title: 'Build a date picker' },
+    { id: '2', state: 'TASK_INBOX', title: 'QA dropdown' },
+    {
+      id: '3',
+      state: 'TASK_INBOX',
+      title: 'Write a schema for account avatar component',
+    },
+    { id: '4', state: 'TASK_INBOX', title: 'Export logo' },
+    { id: '5', state: 'TASK_INBOX', title: 'Fix bug in input error state' },
+    { id: '6', state: 'TASK_INBOX', title: 'Draft monthly blog to customers' },
+  ],
+};
+
+export const WithPinnedTasks = Template.bind({});
+WithPinnedTasks.args = {
+  tasks: [
+    { id: '6', title: 'Draft monthly blog to customers', state: 'TASK_PINNED' },
+    ...Default.args.tasks.slice(0, 5),
+  ],
+};
+
+export const Loading = Template.bind({});
+Loading.args = {
+  tasks: [],
+  loading: true,
+};
+
+export const Empty = Template.bind({});
+Empty.args = {
+  ...Loading.args,
+  loading: false,
+};
+```
+
+Shaping stories through args composition is a powerful technique. It allows us to write stories without repeating the same data over and over again. And more importantly, it tests component integration. If you rename one of the `Task` component props, that'll lead to failed test cases for `TaskList`.
+
+![](/ui-testing-handbook/tasklist-stories.gif)
+
+So far, we’ve only dealt with components that accept data and callback via props. Things get trickier when your component is wired up to an API or has internal state. Next we'll look at how to isolate and test such connected components.
+
+### Stateful composite components
+
+The `InboxScreen` uses a [custom hook](https://github.com/chromaui/ui-testing-guide-code/blob/composition-testing/src/useTasks.js) to fetch data from the Taskbox API and to manage application state. Much like unit tests, we want to detach components from the real backend and test the features in isolation.
+
+![](/ui-testing-handbook/taskbox.png)
+
+That’s where Storybook addons come in. They allow you to mock API requests, state, context, providers and anything else that your component relies on. Teams at [The Guardian](https://5dfcbf3012392c0020e7140b-borimwnbdl.chromatic.com/?path=/story/layouts-showcase--article-story) and [Sidewalk Labs](https://www.sidewalklabs.com/) (Google) use them to build entire pages in isolation.
+
+For the InboxScreen, we are going to use [Mock Service Worker (MSW) addon](https://storybook.js.org/addons/msw-storybook-addon/) to intercept requests at the network level and return mocked responses.
+
+Install msw & its storybook addon.
+
+```
+yarn add -D msw msw-storybook-addon
+```
+
+Then, generate a new service worker in your public folder.
+
+```
+npx msw init public/
+```
+
+Enable the MSW addon in Storybook by adding this to your `.storybook/preview.js` file:
+
+```javascript:title=.storybook/preview.js
+import { addDecorator } from '@storybook/react';
+import { initialize, mswDecorator } from 'msw-storybook-addon';
+
+initialize();
+addDecorator(mswDecorator);
```

Comment:
Small thing, but our [decorators docs](https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators) recommend the `export const decorators = [mswDecorator]` syntax.